internal/server/api.go

```diff
@@ -18,64 +18,79 @@ package server
 
 import (
 	"context"
-	"net/http"
 	"strings"
 
 	"github.com/thestormforge/optimize-controller/v2/internal/version"
 	"github.com/thestormforge/optimize-go/pkg/api"
+	applications "github.com/thestormforge/optimize-go/pkg/api/applications/v2"
 	experimentsv1alpha1 "github.com/thestormforge/optimize-go/pkg/api/experiments/v1alpha1"
 	"github.com/thestormforge/optimize-go/pkg/config"
 )
 
 const audience = "https://api.carbonrelay.io/v1/"
 
 func NewExperimentAPI(ctx context.Context, uaComment string) (experimentsv1alpha1.API, error) {
-	address, rt, err := loadConfig(ctx, uaComment)
+	client, err := newClientFromConfig(ctx, uaComment)
 	if err != nil {
 		return nil, err
 	}
 
-	// Create a new Experiment API client
-	c, err := api.NewClient(address, rt)
+	expAPI := experimentsv1alpha1.NewAPI(client)
+
+	// An unauthorized error means we will never be able to connect without changing the credentials and restarting
+	if _, err := expAPI.CheckEndpoint(ctx); api.IsUnauthorized(err) {
+		return nil, err
+	}
+
+	return expAPI, nil
+}
+
+func NewApplicationAPI(ctx context.Context, uaComment string) (applications.API, error) {
+	client, err := newClientFromConfig(ctx, uaComment)
 	if err != nil {
 		return nil, err
 	}
 
-	expAPI := experimentsv1alpha1.NewAPI(c)
+	appAPI := applications.NewAPI(client)
 
-	// An unauthorized error means we will never be able to connect without changing the credentials and restarting
-	if _, err := expAPI.Options(ctx); api.IsUnauthorized(err) {
+	if _, err := appAPI.CheckEndpoint(ctx); api.IsUnauthorized(err) {
 		return nil, err
 	}
 
-	return expAPI, nil
+	return appAPI, nil
 }
 
-func loadConfig(ctx context.Context, uaComment string) (string, http.RoundTripper, error) {
+func newClientFromConfig(ctx context.Context, uaComment string) (api.Client, error) {
 	// Load the configuration
 	cfg := &config.OptimizeConfig{}
 	cfg.AuthorizationParameters = map[string][]string{
 		"audience": {audience},
 	}
 
 	if err := cfg.Load(); err != nil {
-		return "", nil, err
+		return nil, err
 	}
 
 	// Get the Experiments API endpoint from the configuration
 	// NOTE: The current version of the configuration has an explicit configuration for the
 	// experiments endpoint which would duplicate the "/experiments/" path segment
 	srv, err := config.CurrentServer(cfg.Reader())
 	if err != nil {
-		return "", nil, err
+		return nil, err
 	}
 
-	address := strings.TrimSuffix(srv.API.ExperimentsEndpoint, "/v1/experiments/")
+	address := strings.TrimSuffix(srv.API.ApplicationsEndpoint, "/v2/applications/")
```

Comment:
This still doesn't address my original point: the way we are setup right now you should not use the same client for the application and experiment APIs. The least impactful hack around this that I can think of right now would be something like this:
```
func newClientFromConfig(ctx context.Context, uaComment string, address func(config.Server) string) (api.Client, error) {
  ...
  c, err := api.NewClient(address(srv), rt)
  ...
}

# From NewExperimentAPI call:
client, err := newClientFromConfig(ctx, uaComment, func(srv configServer) string {
  return strings.TrimSuffix(srv.API.ExperimentsEndpoint, "/v1/experiments/")
})

# From NewApplicationAPI call:
client, err := newClientFromConfig(ctx, uaComment, func(srv configServer) string {
  return strings.TrimSuffix(srv.API. ApplicationsEndpoint, "/v2/applications/")
})
```

Reply:
they're still different http clients, just both configured to the same endpoint.
but do we want to have them split? I thought since we'll be proxying requests through applications endpoint shouldn't they be using the same endpoint?